File: apps/api/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    name: str
    address: str | None = None
    pnu: str | None = None
    description: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = None
    status: str | None = None
    risk_level: str | None = None
    address: str | None = None
    pnu: str | None = None
    description: str | None = None


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    owner_id: str | None = None
    name: str
    status: str
    risk_level: str | None = None
    address: str | None = None
    pnu: str | None = None
    description: str | None = None
    created_at: datetime
    updated_at: datetime

File: apps/web/src/types/project.ts
export interface Project {
  id: string;
  name: string;
}

export interface ApiProject {
  id: string;
  owner_id: string | null;
  name: string;
  status: string;
  risk_level: string | null;
  address: string | null;
  pnu: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectRow {
  id: string;
  name: string;
  address: string;
  pnu: string;
  status: string;
  updatedAt: string;
  risk: string;
  action: string;
}
